app/routes/goal_routes.py

```diff
@@ -1 +1,136 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, make_response, abort, request
+
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+
+@goal_bp.post("")
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body.keys():
+        abort(make_response({"details": "Invalid data"}, 400))
+    title = request_body["title"]
+    new_goal = Goal(title=title)
+    db.session.add(new_goal)
+    db.session.commit()
+    response = {
+        "goal": new_goal.to_dict()
+    }
+    return response, 201
+
+
+@goal_bp.post("/<goal_id>")
+def create_goal_with_id(goal_id):
+    request_body = request.get_json()
+    if "title" not in request_body.keys():
+        abort(make_response({"details": "Invalid data"}, 400))
+    title = request_body["title"]
+    new_goal = Goal(id=goal_id, title=title)
+    db.session.add(new_goal)
+    db.session.commit()
+    response = {
+        "goal": new_goal.to_dict()
+    }
+    return response, 201
+
+
+@goal_bp.get("")
+def get_all_tasks():
+    title_param = request.args.get("title")
+
+    query = db.select(Goal)
+    if title_param:
+        query = query.where(Goal.title.like(f"%{title_param}"))
+
+    goals = db.session.scalars(query).all()
+    goals_response = [goal.to_dict() for goal in goals]
+    return goals_response
```

Comment:
Notice that you instantiate the variable `goals_response`, but then the variable doesn't get referenced or changed before you return it on the next line. You can just return the list without creating a variable.
```suggestion
    return [goal.to_dict() for goal in goals]
```